packages/schema-poller/src/neo4jConnection.ts

```diff
@@ -8,10 +8,21 @@ const METADATA_BASE = {
   version: version,
 };
 
-function resolveInitialDatabase(databases: Database[]): string {
+function resolveInitialDatabase(
+  databases: Database[],
+  database?: string,
+): string {
+  if (database) {
+    const resolvedDatabase = databases.find((d) => d.name === database);
+    if (resolvedDatabase) {
+      return resolvedDatabase.name;
+    }
+  }
+
   const home = databases.find((d) => d.home);
   const def = databases.find((d) => d.default);
   const system = databases.find((d) => d.name === 'system');
+
   return home?.name ?? def?.name ?? system?.name ?? databases[0]?.name;
```

Comment:
Should we have this logic anymore? If we set an explicit connection to a specific database that does not exist, would this try to connect to the home one as a fallback? Maybe it's not what a user expects

Reply:
I think the current db should be set to undefined in that case and the healthcheck return false?

Reply:
yeah - that makes a lot of sense, have updated